pkg/messaging/v1/invoke_method_request.go

```diff
@@ -104,6 +104,23 @@ func (imr *InvokeMethodRequest) WithHTTPExtension(verb string, querystring strin
 	return imr
 }
 
+// AppendMetadata appends to existing metadata
+func (imr *InvokeMethodRequest) AppendMetadata(md map[string][]string) *InvokeMethodRequest {
+	metadata := imr.r.Metadata
+
+	for k, values := range md {
+		listValue := internalv1pb.ListStringValue{}
+		if _, ok := metadata[k]; ok {
+			listValue = *metadata[k]
+		}
+		listValue.Values = append(listValue.Values, values...)
+		metadata[k] = &listValue
```

Comment:
I would rewrite this like below
```
for k, values := range md {
   if _, ok:=metadata[k]; !ok {
     metadata[k] = &internalv1pb.ListStringValue{Values:[]string{}}
   }
   metadata[k].Values = append(metadata[k].Values, values...)
}
```

Reply:
That's better, made a small change though : Merged all values into a single string here itself because `Forwarded` header is first set in useProxy, remaining values are appended in invokeRemote